lib/bundler/compact_index_client/updater.rb

```diff
@@ -26,58 +26,57 @@ def initialize(fetcher)
       end
 
       def update(local_path, remote_path, retrying = nil)
-        headers = {}
-
-        Dir.mktmpdir("bundler-compact-index-") do |local_temp_dir|
-          local_temp_path = Pathname.new(local_temp_dir).join(local_path.basename)
-
-          # first try to fetch any new bytes on the existing file
-          if retrying.nil? && local_path.file?
-            FileUtils.cp local_path, local_temp_path
-            headers["If-None-Match"] = etag_for(local_temp_path)
-            headers["Range"] =
-              if local_temp_path.size.nonzero?
-                # Subtract a byte to ensure the range won't be empty.
-                # Avoids 416 (Range Not Satisfiable) responses.
-                "bytes=#{local_temp_path.size - 1}-"
-              else
-                "bytes=#{local_temp_path.size}-"
-              end
-          else
-            # Fastly ignores Range when Accept-Encoding: gzip is set
-            headers["Accept-Encoding"] = "gzip"
-          end
-
-          response = @fetcher.call(remote_path, headers)
-          return nil if response.is_a?(Net::HTTPNotModified)
+        local_temp_dir = create_tmpdir
```

Comment:
moving away from the block form to this will no longer clean up the tmpdir, can we ensure that we remove the directory if it exists?

Reply:
In that case I think it would be best if I still called ```Dir.mktmpdir("bundler-compact-index-")``` with the block, and just rescued ```Errno::EACCES``` within the ```update``` method if the call to ```Dir.mktmpdir``` raises an exception.

Reply:
sure